fusion/dsl/README.md

```diff
@@ -0,0 +1,205 @@
+8.    g4 = g5 = g7 = g8 = g9    // C2
+9.    g1 != g4                  // C3, the contrived constraint
+10.    c3 = c4                  // C4
+11.  }                          // C5 and C6 do not require a statement
+```
+
+Aspects of the syntax
+  -  Sequence and constraint sections can be anonymous or named
+    -  In the above sample the constraints are anonymous and the sequence has a declared identifier seq1
+  -  The section declaration syntax, sequence{} and constraints{}, is optional
+  -  The statements within a sequence must be in the correct order
+    -  A sequence with relaxed adjacency requirements uses two key words to identify gaps in the sequence, see Section 6.4
+  -  The sequence statements must appear before any of the statements associated with the constraints
+  -  The constraints statements must occur before any statements associated with transforms.
+
+Syntax for simple assignment is also proposed. For example:
+```
+g1 = x13
+```
+
+## 5.4  Source using explicit constraints
+Below is an example declaration of a fusion operation that uses explicit constraints between operands uses
+```
+1.  fusion_decl fs1 {
+2.    sequence seq {
+3.      c.lui    g1, c1
+4.      c.addi   g2, g3, c2
+5.      c.xor    g4, g5, g6
+6.      c.slli   g7, g8, c3
+7.      c.srli   g9,    c4
+8.    }
+9.  
+10.    constraints cons {  
+11.      gpr g1,g2,g3,g4,g5,g6,g7,g8,g9
+12.      s20 c1
+13.      s12 c2
+14.      u6  c3,c4
+15.  
+16.      g1 = g2 = g3 = g6       // This expresses C1
+17.      g4 = g5 = g7 = g8 = g9  // C2
+18.      g1 != g4                // C3, the contrived constraint
+19.      c3 = c4                 // C4
+20.    }
+21.  
+22.    transform t1.seq.cons     // Still working on the transform syntax
+23.                              // this statement is a reference to an 
+24.                              // externally described transform
+25.  }
+```
+## 5.5  Source using positional constraints
+The above can be simplified by redefining the variables and using the implicit constraints of the sequence operand positions. G[1,2,3,6] are replaced with g1, g[4,5,7,8,9] are replaced by g2. The result is that the constraints can be simplified as well.
+```
+1.  fusion_decl fs1 {
+2.    sequence {                // Sequence using positional constraints
+3.      0xb      g1, c1         // I’ve shown the alternative syntax where
+4.      c.addi   g1, g1, c2     // the mnemonic is replaced with the Mavis
+5.      c.xor    g2, g2, g1     // UID, for the c.lui equiv function
+6.      c.slli   g2, g2, c3     
+7.      c.srli   g2,     c3
+8.    }
+9.  
+10.    gpr g1,g2
+11.    s20 c1
+12.    s12 c2
+13.    u6 c3
+14.  
+15.    //The constraints declarations
+16.    g1 != g2                 // C3, the contrived constraint
+17.                             // In this version of the syntax the
+18.    transform t1             // scope of sequence and constraints is
+19.                             // within this fusion_decl
```

Comment:
Is this missing the `constraints` keyword?

Reply:
Yes, this text is old, needs an update.

Reply:
I've remove this file for now. When we've had time to discuss the syntax/grammar there would be an update to this text.